akka-docs/rst/scala/http/client-side/websocket-support.rst

```diff
@@ -3,6 +3,61 @@
 Client-Side WebSocket Support
 =============================
 
-Not yet implemented see 17275_.
+Client side WebSocket support is available through ``Http().singleWebSocketRequest()`` ,
+``Http().webSocketClientFlow()`` and ``webSocketClientLayer``. As a WebSocket consists of two flows of
+messages of which either might happen first, rather than the request and response of a regular HTTP request,
+WebSocket requests always works with ``Flow[Message, Message, Mat]` where the messages will pass through.
+
+A WebSocket request starts with a regular HTTP request which contains an ``Upgrade`` header, so in
+addition to the flow of messages there also is an initial response from the server, this is modelled
+with ``WebSocketUpgradeResponse``.
+
```

Comment:
Right, it's perhaps good to explain somewhere that the upgrade to WS connection is handled automatically by the exposed methods we provide and then the given flow is used to handle the WS communication